content/docs/guides/adopting/from_aws.md

```diff
@@ -0,0 +1,567 @@
+
+{{< langchoose csharp >}}
+
+```javascript
+let aws = require("@pulumi/aws");
+
+let vpc = new aws.ec2.Vpc("myVpc", {
+    cidrBlock: "10.0.0.0/16",
+    tags: { Name: "Primary_CF_VPC" },
+}, { import: "vpc-0e1a74859af1da17f" });
+
+module.exports = {
+    vpcId: vpc.id,
+};
+```
+
+```typescript
+import * as aws from "@pulumi/aws";
+
+const vpc = new aws.ec2.Vpc("myVpc", {
+    cidrBlock: "10.0.0.0/16",
+    tags: { Name: "Primary_CF_VPC" },
+}, { import: "vpc-0e1a74859af1da17f" });
+
+export const vpcId = vpc.id;
+```
+
+```python
+import pulumi_aws as aws
+
+vpc = aws.ec2.Vpc('myVpc',
+    cidr_block='10.0.0.0/16',
+    tags={ 'Name': 'Primary_CF_VPC' },
+    opts=ResourceOptions(import_='vpc-0e1a74859af1da17f')
+)
+
+pulumi.export('vpc_id', vpc.id)
+```
+
+```go
+package main
+
+import (
+    "github.com/pulumi/pulumi-aws/sdk/go/aws/ec2"
+    "github.com/pulumi/pulumi/sdk/go/pulumi"
+)
+
+func main() {
+    pulumi.Run(func(ctx *pulumi.Context) error {
+        vpc, err := ec2.NewVpc(ctx, "myVpc",
+            &ec2.VpcArgs{
+                CidrBlock: pulumi.String("10.0.0.0/16"),
+                Tags: pulumi.Map{
+                    "Name": pulumi.String("Primary_CF_VPC"),
+                },
+            },
+            pulumi.Import(pulumi.ID("vpc-0e1a74859af1da17f")),
+        })
+        if err != nil {
+            return err
+        }
+
+        ctx.Export("vpcId", vpc.ID())
+        return nil
+    })
+}
+```
+
+```csharp
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+using Pulumi;
+using Ec2 = Pulumi.Aws.Ec2;
+
+class Program
+{
+    static Task<int> Main()
+    {
+        return Deployment.RunAsync(() => {
+            var vpc = new Ec2.Vpc("myVpc",
+                new Ec2.VpcArgs
+                {
+                    CidrBlock = "10.0.0.0/16",
+                    Tags = new Dictionary<string, object?>
+                    {
+                        { "Name", "Primary_CF_VPC" },
+                    },
+                },
+                new CustomResourceOptions {
+                    ImportId = "vpc-0e1a74859af1da17f",
+                },
+            );
+
+            return new Dictionary<string, object?>
+            {
+                { "vpcId", vpc.Id },
+            };
+        });
+    }
+}
+```
+
+After running `pulumi up`, your VPC will become under the control of Pulumi without any disruption, and you can then delete the import directives from your code. All subsequent infrastructure changes you'd like to be made can happen within Pulumi instead of AWS CloudFormation.
```

Comment:
There's an implied, but not explicit as far as I can tell, step here of removing the cloudformation template from the `.ts` code so that it gets deleted, but the resource doesn't experience any disruption.  Might be worth showing the `pulumi up` that does the transition?  (imports the resource, then destroys the cloudformation stack).

Reply:
The idea here was that because of the retain-upon-deletion annotation on the CloudFormation resource, these don't need to be split into two distinct steps. Do you think that would help with clarity here?

Reply:
My point was just to make the one step more clear.  That the one step both imports the resource and destroys the CloudFormation stack (without interrupting the actual cloud resources you care about).